File: backend/src/main/java/edu/gsu/pantherwatch/pantherwatch/api/MeetingTime.java
package edu.gsu.pantherwatch.pantherwatch.api;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class MeetingTime {
    @JsonProperty("beginTime")
    private String beginTime;

    @JsonProperty("buildingDescription")
    private String buildingDescription;

    @JsonProperty("campusDescription")
    private String campusDescription;

    @JsonProperty("category")
    private String category;

    @JsonProperty("endDate")
    private String endDate;
    
    @JsonProperty("endTime")
    private String endTime;

    @JsonProperty("sunday")
    private boolean sunday;

    @JsonProperty("monday")
    private boolean monday;

    @JsonProperty("tuesday")
    private boolean tuesday;

    @JsonProperty("wednesday")
    private boolean wednesday;

    @JsonProperty("thursday")
    private boolean thursday;

    @JsonProperty("friday")
    private boolean friday;

    @JsonProperty("saturday")
    private boolean saturday;

    @JsonProperty("hoursWeek")
    private int hoursWeek;

    @JsonProperty("meetingTypeDescription")
    private String meetingTypeDescription;

    @JsonProperty("room")
    private String room;

    @JsonProperty("startDate")
    private String startDate;
}

File: frontend-android/app/src/main/java/edu/gsu/pantherwatch/data/model/MeetingTime.kt
package edu.gsu.pantherwatch.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class MeetingTime(
    @field:Json(name = "beginTime")
    val beginTime: String?,
    
    @field:Json(name = "buildingDescription")
    val buildingDescription: String?,
    
    @field:Json(name = "campusDescription")
    val campusDescription: String?,
    
    @field:Json(name = "category")
    val category: String?,
    
    @field:Json(name = "endDate")
    val endDate: String?,
    
    @field:Json(name = "endTime")
    val endTime: String?,
    
    @field:Json(name = "friday")
    val friday: Boolean,
    
    @field:Json(name = "hoursWeek")
    val hoursWeek: Int,
    
    @field:Json(name = "meetingTypeDescription")
    val meetingTypeDescription: String?,
    
    @field:Json(name = "monday")
    val monday: Boolean,
    
    @field:Json(name = "room")
    val room: String?,
    
    @field:Json(name = "saturday")
    val saturday: Boolean,
    
    @field:Json(name = "startDate")
    val startDate: String?,
    
    @field:Json(name = "sunday")
    val sunday: Boolean,
    
    @field:Json(name = "thursday")
    val thursday: Boolean,
    
    @field:Json(name = "tuesday")
    val tuesday: Boolean,
    
    @field:Json(name = "wednesday")
    val wednesday: Boolean
)
